### DISCUSSION TITLE
config/sidebar/

### DISCUSSION THREAD

**Discussion by giscus** [pengzhanbo/vuepress-theme-plume#142] [docs-comment]
# config/sidebar/

概述 在本主题中，侧边栏 特指 位于页面 最左侧的内容区域，用于导航到不同的页面。 在 vuepress 的默认主题 @vuepress/theme-default 中，侧边栏是通过 sidebar 进行配置。 但在本主题中，为使其更具语义化，主题将侧边栏配置整合到了 中 。 通过一个 note 对应一个 sidebar 的方式，实现更加简洁，语义化的...

https://plume.pengzhanbo.cn/config/sidebar/

**Comment by tianhairui** [comment]
我在按照 `Notes 中的侧边栏配置` 章节配置`config.js`的时候，保存后，提示该模块未提供，提示如下：
```shell file:///E:/blog/my-blog/docs/.vuepress/config.js.1ae59116.mjs:4
import { defineNoteConfig, plumeTheme } from "vuepress-theme-plume";
         ^^^^^^^^^^^^^^^^
SyntaxError: The requested module 'vuepress-theme-plume' does not provide an export named 'defineNoteConfig'
    at ModuleJob._instantiate (node:internal/modules/esm/module_job:134:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:217:5)
    at async ModuleLoader.import (node:internal/modules/esm/loader:316:24)
    at async loadUserConfig (file:///E:/blog/my-blog/node_modules/@vuepress/cli/dist/index.js:77:18)
    at async CAC.dev (file:///E:/blog/my-blog/node_modules/@vuepress/cli/dist/index.js:443:52)
```
我在文档中也未搜索到关于`defineNoteConfig`的内容，请问是因为还没有添加？还是我未开启相关模块功能？

**Reply by pengzhanbo** [reply]
在新版本 `1.0.0-rc.87` 中的更新，替换了函数名，本来想周六早上更新的，但我有其它事情耽搁到了现在 😂

**Comment by zhengchen-zzZ** [comment]
作者，我想某个 `note` 如果是配置自动生成侧边栏，每项配置`sidebar: 'auto'`，有没有配置可以即自动生成侧边栏，又可以方便控制侧边栏项折叠状态的方法。

有些文档为了分类方便，所以用文件夹管理嵌套笔记的比较多，在侧边栏展示的时候会比较长而且比较奇怪。虽说可以通过手动配置达到折叠效果，设置collapse完成这个需求，但要手动配置侧边栏还是麻烦。所以有什么好的解决办法吗。

**Reply by pengzhanbo** [reply]
暂时没有什么好的方法，这是一个权衡取舍，`auto` 方便的同时也会失去配置的灵活度。就像排序，不得不需要在 文件夹和文件名前 加序号。虽然可以采用相同的策略，在 文件夹名后面跟上某些特殊字符来决定展开/折叠。 但文件夹名本身在不同系统就有一堆不可使用的非法字符，要在键盘上找到有代表性的接受度高的容易键入的合法字符，并不是件容易的事。

**Comment by Adoubf** [comment]
大佬，帮我看看是什么设置的问题，我的目录树如下，最外层的2个文件夹放在notes里，但是启用自动生成侧边栏后，并没有生成侧边栏
01-文件夹
- 01-文件夹
-- asset文件夹
-- 00-md文件
-- 01-md文件
-- 02-md文件
-- ......
-- 0[n]-md文件
- 02-文件夹
-- asset文件夹
-- 00-md文件
-- 01-md文件
-- 02-md文件
-- ......
-- 0[n]-md文件
- ....
- 0[n]-文件夹

02-文件夹
- asset文件夹
- 00-md文件
- 01-md文件
- 02-md文件
- ......
- 0[n]-md文件

**Reply by pengzhanbo** [reply]
这些信息无助于解决问题，请提供目录结构截图，以及相关配置信息的截图。

**Comment by BoogonClothman** [comment]
在不使用notes的情况下，通用配置侧边栏后，在目录的README页面正常显示，但是进入具体文章时，侧边栏消失，这是什么原因？

**Reply by pengzhanbo** [reply]
我无法从你给出的信息中推测原因，如果方便，请给出你的仓库地址，或者把相关的配置和目录结构发一下。

**Reply by c2ccc2** [reply]
我也碰到了同样的问题。
简单来说就是配置了note后，点击note中的某一章节，进入的详情页没有侧边栏。

**Comment by HMTeen** [comment]
作者大佬，有没有能够修改侧边栏宽度的接口？
想让左侧边栏，中间内容页，右边目录页的比例可调节，更灵活一点

**Reply by pengzhanbo** [reply]
可以通过 css vars `--vp-sidebar-width` 控制。

**Comment by Sunny19191** [comment]
该主题是否支持收缩/展示侧边栏功能？或者是有没有自定义开发的位置来将该功能集成进去？

**Reply by pengzhanbo** [reply]
[查看文档](https://plume.pengzhanbo.cn/guide/document/#%E8%87%AA%E5%AE%9A%E4%B9%89%E4%BE%A7%E8%BE%B9%E6%A0%8F) ，通过 `collapsed` 配置

**Reply by Sunny19191** [reply]
不好意思，不是折叠/展开文件夹，是收缩/展开整理侧边栏功能，收缩侧边栏会自动扩大文档页面宽度，类似如下：
<img width="328" height="967" alt="image" src="https://github.com/user-attachments/assets/3d094b26-937e-4175-a802-bc2e6e572d36" />
